src/main/java/de/hhu/propra/link/controllers/LinkController.java

```diff
@@ -39,6 +39,12 @@ public String index(Model model) {
     public String newLink(@ModelAttribute @Valid Link link, BindingResult bindingResult) {
         this.currentLink = link;
 
+        // suggest an abbreviation if not set
+        if(link.getAbbreviation().isEmpty()){
+            String abbrevation = makeAbbrevation(new String(link.getUrl()));
+            link.setAbbreviation(abbrevation);
+        }
+
```

Comment:
This leaves the scope of a real controller. We should create a `LinkService` for this instead.

Reply:
Also, please make a function from it. Don't add comments when they are not necessary. And it is often a better practice to define a function when you have the need to write a comment.